apps/api/routes.py

```diff
@@ -194,6 +194,30 @@ def save_lab_answers(lab_inst_id):
 
     return {"status": "ok", "result": "Answers saved successfully"}, 200
 
+@blueprint.route('/lab_answers/correct/<int:answer_id>', methods=["PUT"])
+@login_required
+def correct_lab_answer(answer_id):
+    if current_user.category == "user" or current_user.category == "student":
+        return {}, 401
```

Comment:
Nice that you added extra validation here. But it is still missing the check to validate if the teacher is allowed to update this particular lab answer.